src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesSubTreeProviderBase.cs

```diff
@@ -227,13 +249,26 @@ protected override void Dispose(bool disposing)
             // process separatelly shared projects changes
             ProcessSharedProjectImportNodes(e.Value.Item3, dependenciesChange);
 
-            // Apply dependencies changes to actual RootNode children collection
-            // remove first nodes from actual RootNode
-            dependenciesChange.RemovedNodes.ForEach(RootNode.RemoveChild);
+            lock (_rootNodeSync)
+            {
+                // Apply dependencies changes to actual RootNode children collection
+                // remove first nodes from actual RootNode
+                dependenciesChange.RemovedNodes.ForEach(RootNode.RemoveChild);
+
+                ProcessDuplicatedNodes(dependenciesChange);
 
-            ProcessDuplicatedNodes(dependenciesChange);
+                dependenciesChange.UpdatedNodes.ForEach((topLevelNode) =>
+                {
+                    var oldNode = RootNode.Children.FirstOrDefault(x => x.Id.Equals(topLevelNode.Id));
```

Comment:
its under lock

---
In reply to: [85814342](https://github.com/dotnet/roslyn-project-system/pull/675#discussion_r85814342) [](ancestors = 85814342)